server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -501,23 +523,38 @@ private void updateCleanupKeyToCountMapOnCacheEviction(CleanupKey cleanupKey) {
             }
             ShardId shardId = indexShard.shardId();
 
-            cleanupKeyToCountMap.computeIfPresent(shardId, (shard, keyCountMap) -> {
-                keyCountMap.computeIfPresent(cleanupKey.readerCacheKeyId, (key, currentValue) -> {
-                    // decrement the stale key count
-                    staleKeysCount.decrementAndGet();
-                    int newValue = currentValue - 1;
-                    // Remove the key if the new value is zero by returning null; otherwise, update with the new value.
+            cleanupKeyToCountMap.computeIfPresent(shardId, (id, stringIntegerConcurrentMap) -> {
```

Comment:
lets rename `stringIntegerConcurrentMap` to `readerKeyToCount`

Reply:
When shard is closed then there won't be any entries in the `cleanupKeyToCountMap` and we will need to decrement the `staleKeysCount`